design_docs/analysis_task_template.md

```diff
@@ -0,0 +1,594 @@
+funciton would look like:
+```python
+def check(config, streamMap=None, variableMap=None, fieldName=None,
+          **kwargs):  # {{{
+    """
+    Checks whether the configuration is correct for a given analysis.  For
+    example, checks if necessary observations and other data files are found.
+
+    Parameters
+    ----------
+    config :  instance of MpasAnalysisConfigParser
+        Contains configuration options used to customize the analysis.
+
+    streamMap : dict, optional
+        A dictionary of MPAS component stream names that map to their
+        MPAS-Analysis counterparts.  These are used to maintain compatibility
+        across various versions of MPAS components where stream names have
+        changed.
+
+    variableMap : dict, optional
+        A dictionary of MPAS component variable names that map to their
+        MPAS-Analysis counterparts. These are used to maintain compatibility
+        across various versions of MPAS components where variable names have
+        changed.
+
+    fieldName : {'sst', 'mld', 'sss'}
+        The name of the field to be analyzed
+
+    kwargs : dict
+        A dictionary of remaining keyword arguments (ignored in this function)
+
+    Raises
+    ------
+    ValueError
+        If fieldName was not supplied
+    OSError
+        If files are not present
+
+    Authors
+    -------
+    Xylar Asay-Davis
+
+    Last Modified
+    -------------
+    03/08/2017
+    """
+
+    possibleFieldNames = ['sst', 'mld', 'sss']
+    if fieldName not in possibleFieldNames:
+        raise ValueError('fieldName must be one of {}.'.format(
+            possibleFieldNames))
+
+    section = 'climatology'
+    startDate = '{:04d}-01-01_00:00:00'.format(
+        config.getint(section, 'startYear'))
+    if not config.has_option(section, 'startDate'):
+        config.set(section, 'startDate', startDate)
+    endDate = '{:04d}-12-31_23:59:59'.format(
+        config.getint(section, 'endYear'))
+    if not config.has_option(section, 'endDate'):
+        config.set(section, 'endDate', endDate)
+
+    if config.get('runs', 'preprocessedReferenceRunName') != 'None':
+            check_path_exists(config.get('oceanPreprocessedReference',
+                                         'baseDirectory'))
+
+    make_directories(build_config_full_path(config, 'output',
+                                            'plotsSubdirectory'))
+
+    return  # }}}
+```
+This code has been taken out of `run_analysis.py`, simplifying and clarifying
+the code.
+
+<h4> Implementation: Analysis Continues even when Analysis Task Fails <br>
+Date last modified: 2017/03/08 <br>
+Contributors: Xylar Asay-Davis
+</h4>
+
+Calls to `check` and `run` functions in `run_analysis.py` are inside of 
+`try/except` blocks, which catch the exceptions and print the stack trace 
+but don't cause the code to exit.
+```python
+try:
+    analysisModule.check(config, **kwargs)
+    add = True
```

Comment:
This snippit of code is part of a larger file.  `add = True` makes sense in the larger code.  I've replaced it by `...` here so it won't confuse the reader.